pages/previewModel.js

```diff
@@ -86,11 +115,20 @@ function loadModel(gltf, resetCamera) {
 
         viewer._model = model;
         viewer.onLoad.raiseEvent(model);
-    }).otherwise(function(e){
+    }).otherwise(function(e) {
         window.alert('Error: ' + e);
     });
 }
 
-var gltf = JSON.parse(document.getElementById('glTF').textContent);
+var gltfFileName = document.getElementById('gltfFileName').textContent;
+var gltfRootPath = document.getElementById('gltfRootPath').textContent;
 
-loadModel(gltf, true);
+try {
+    var gltfContent = JSON.parse(document.getElementById('gltf').textContent);
+    loadModelFromContent(gltfContent, gltfRootPath, true);
+}
+catch (ex) {
+    // If the glTF content is missing or not valid JSON, then try to load the
+    // model directly from the glTF file.
+    loadModelFromFile(gltfFileName, gltfRootPath, true);
```

Comment:
So, the main reason I was adding support for loading via file, is because that's how the other three Renderers will be working.  Cesium appears to be the only one that has the option to take in the raw text of the gltf file and use that as an object....the other three (babylon.js, three.js and osg.js) only take in uri's to the files.  I can add an extra div that hides in say, 5 seconds, that just lets the user know that it is using the last saved file content instead of the raw file.